mmseg/datasets/pipelines/transforms.py

```diff
@@ -463,6 +465,61 @@ def __repr__(self):
         return self.__class__.__name__ + f'(crop_size={self.crop_size})'
 
 
+@PIPELINES.register_module()
+class Rgb2Gray(object):
+    """Convert RGB image to grayscale image.
+
+    This transform calculate the weighted mean of input image channels with
+    ``weights`` and then expand the channels to ``out_channels``. When
+    ``out_channels`` equals 0, the number of output channels is the same as
+    input channels.
+
+    Args:
+        out_channels (int): Expected number of output channels after
+            transforming.
+        weights (tuple[float]): The weights to calculate the weighted mean.
+            Default: (0.299, 0.587, 0.114).
+    """
+
+    def __init__(self, out_channels, weights=(0.299, 0.587, 0.114)):
```

Comment:
Why using weighted sum instead of direct `mmcv.rgb2gray`.

Reply:
By this method, we can use the weights defined by us instead of fixed weights.